runtime/lua/vim/_editor.lua

```diff
@@ -1036,56 +1036,97 @@ end
 ---@param version     string     Version when the deprecated function will be removed.
----@param plugin      string|nil Name of the plugin that owns the deprecated feature.
----                              Defaults to "Nvim".
+---@param plugin      string     Name of the plugin that owns the deprecated feature.
 ---@param backtrace   boolean|nil Prints backtrace. Defaults to true.
 ---
 ---@return string|nil # Deprecated message, or nil if no message was shown.
 function vim.deprecate(name, alternative, version, plugin, backtrace)
-  plugin = plugin or 'Nvim'
-  local will_be_removed = 'will be removed'
-
-  -- Only issue warning if feature is hard-deprecated as specified by MAINTAIN.md.
-  -- Example: if removal_version is 0.12 (soft-deprecated since 0.10-dev), show warnings starting at
-  -- 0.11, including 0.11-dev
-  if plugin == 'Nvim' then
-    local major, minor = version:match('(%d+)%.(%d+)')
-    major, minor = tonumber(major), tonumber(minor)
-
-    local hard_deprecated_since = string.format('nvim-%d.%d', major, minor - 1)
-    -- Assume there will be no next minor version before bumping up the major version
-    local is_hard_deprecated = minor == 0 or vim.fn.has(hard_deprecated_since) == 1
-    if not is_hard_deprecated then
-      return
-    end
-
-    local removal_version = string.format('nvim-%d.%d', major, minor)
-    will_be_removed = vim.fn.has(removal_version) == 1 and 'was removed' or will_be_removed
-  else
-    vim.validate {
-      name = { name, 'string' },
-      alternative = { alternative, 'string', true },
-      version = { version, 'string', true },
-      plugin = { plugin, 'string', true },
-    }
-  end
+  vim.validate {
+    name = { name, 'string' },
+    alternative = { alternative, 'string', true },
+    version = { version, 'string', true },
+    plugin = { plugin, 'string', true },
+  }
 
   local msg = ('%s is deprecated'):format(name)
   msg = alternative and ('%s, use %s instead.'):format(msg, alternative) or (msg .. '.')
-  msg = ('%s%s\nFeature %s in %s %s'):format(
-    msg,
-    (plugin == 'Nvim' and ' :help deprecated' or ''),
-    will_be_removed,
-    plugin,
-    version
-  )
+  msg = ('%s\nFeature will be removed in %s %s'):format(msg, plugin, version)
   local displayed = vim.notify_once(msg, vim.log.levels.WARN)
   if displayed and backtrace ~= false then
     vim.notify(debug.traceback('', 2):sub(2), vim.log.levels.WARN)
   end
   return displayed and msg or nil
 end
 
+do
+  local function truncated_echo(msg)
+    -- Truncate message to avoid hit-enter-prompt
+    local max_width = vim.o.columns * math.max(vim.o.cmdheight - 1, 0) + vim.v.echospace
+    local msg_truncated = string.sub(msg, 1, max_width)
+    vim.api.nvim_echo({ { msg_truncated, 'WarningMsg' } }, true, {})
+  end
+
+  local notified = false
+
+  function vim._truncated_echo_once(msg)
+    if not notified then
+      truncated_echo(msg)
+      notified = true
```

Comment:
I think https://github.com/neovim/neovim/pull/28778/files#r1606057221 is still applicable. `vim._deprecate` should give warning messages for any subsequent calls after the first invocation.

Kind of re-inventing the wheel like `vim.notify_once`, but:
```suggestion
    if not notified[msg] then
      truncated_echo(msg)
      notified[msg] = true
```

Reply:
Uh, you suggested I use notify which doesn’t work. What am I supposed to do?